src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -76,49 +77,72 @@ const useFetchChannelHook = () => {
-                undefined,
-                anonUserInfo
-              );
-
-              channelList.saveIfLatest(localDb).then(() => {
-                refresh('channelList');
-                resolve(true);
+        if (Platform.OS === 'android') {
+          const channelList = ChannelList.fromChannelAPI(
+            newChannel,
+            type[channelType],
+            undefined,
+            anonUserInfo
+          );
+          channelList.saveIfLatest(localDb).then(() => {
+            refresh('channelList');
+          });
+        } else {
+          queue.addPriorityJob({
+            id: `saveChannelData-${channel?.id}`,
+            priority: QueueJobPriority.MEDIUM,
+            operationLabel: DatabaseOperationLabel.CoreChatSystem_SaveTopicChannel,
+            label: `saveChannelData-${channel?.id}`,
+            task: () => {
+              return new Promise((resolve) => {
+                const channelList = ChannelList.fromChannelAPI(
+                  newChannel,
+                  type[channelType],
+                  undefined,
+                  anonUserInfo
+                );
+                channelList.saveIfLatest(localDb).then(() => {
+                  refresh('channelList');
+                  resolve(true);
+                });
               });
-            });
-          }
-        });
+            }
+          });
+        }
       } catch (e) {
         console.log('error on helperChannelPromiseBuilder', e);
       }
     } else {
       try {
-        queue.addJob({
-          label: `saveChannelData-${channel?.id}`,
-          task: () => {
-            return new Promise((resolve) => {
-              const channelList = ChannelList.fromChannelAPI(
-                newChannel,
-                type[channelType],
-                undefined,
-                anonUserInfo
-              );
-              channelList.saveIfLatest(localDb).then(() => {
-                refresh('channelList');
-                resolve(true);
+        if (Platform.OS === 'android') {
+          const channelList = ChannelList.fromChannelAPI(
+            newChannel,
+            type[channelType],
+            undefined,
+            anonUserInfo
+          );
+          channelList.saveIfLatest(localDb).then(() => {
+            refresh('channelList');
+          });
+        } else {
+          queue.addJob({
+            label: `saveChannelData-${channel?.id}`,
+            task: () => {
+              return new Promise((resolve) => {
+                const channelList = ChannelList.fromChannelAPI(
+                  newChannel,
+                  type[channelType],
+                  undefined,
+                  anonUserInfo
+                );
+                channelList.saveIfLatest(localDb).then(() => {
+                  refresh('channelList');
+                  resolve(true);
+                });
               });
-            });
-          }
-        });
+            }
+          });
+        }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous comment, this block also introduces a platform-specific conditional check. The same considerations apply here as well.

```diff
+         if (Platform.OS === 'android') {
+           const channelList = ChannelList.fromChannelAPI(
+             newChannel,
+             type[channelType],
+             undefined,
+             anonUserInfo
+           );
+           channelList.saveIfLatest(localDb).then(() => {
+             refresh('channelList');
+           });
+         } else {
+           queue.addJob({
+             label: `saveChannelData-${channel?.id}`,
+             task: () => {
+               return new Promise((resolve) => {
+                 const channelList = ChannelList.fromChannelAPI(
+                   newChannel,
+                   type[channelType],
+                   undefined,
+                   anonUserInfo
+                 );
+                 channelList.saveIfLatest(localDb).then(() => {
+                   refresh('channelList');
+                   resolve(true);
+                 });
+                });
+             }
+           });
+         }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->